subsys/net/ip/tcp.c

```diff
@@ -2099,7 +2109,9 @@ NET_CONN_CB(tcp_synack_received)
 	struct net_tcp_hdr *tcp_hdr = proto_hdr->tcp;
 	int ret;
 
-	NET_ASSERT(context && context->tcp);
+	NET_ASSERT(context);
+	k_mutex_lock(&context->lock, K_FOREVER);
```

Comment:
Why is lock stuffed between 2 asserts here?

Reply:
Because `context->tcp` can in theory change if someone else is holding the lock.

Reply:
> Because context->tcp can in theory change if someone else is holding the lock.

Ok, ack. I bet I'd personally add a comment on that.